runner/src/internal.rs

```diff
@@ -191,47 +207,82 @@ pub fn run_step(context: &Context, step: &Step) -> HashSet<StatusResultValue> {
             );
             values
         }
+        Step::WithBackground {
+            name,
+            background,
+            foreground,
+        } => {
+            let context = context.child(name);
+            eprintln!("{} {{", context.prefix);
+
+            let mut running_background = background.run(&context.opt);
+            let background_command = if context.opt.commands || context.opt.dry_run {
+                format!("[{}]", running_background.command)
+            } else {
+                "".to_string()
+            };
+            eprintln!(
+                "{} ⊢ {} (background) ...",
+                context.prefix, background_command
+            );
+
+            // Small delay to make sure the background process started.
+            std::thread::sleep(std::time::Duration::from_millis(2000));
```

Comment:
Are we sure 2000 millis is safe for all cases? I guess it would be nice to to have confirmation that the background process started, but if this works it works.

Reply:
No that's just arbitrary, I don't think there is a foolproof way to detect when a server is up, perhaps we could try and connect to the port and poll until that works? Not sure it's worth, but perhaps it's simpler than it sounds, being in Rust as opposed to Bash? Future work material :)

Reply:
Yeah I figured it wouldn't be as easy as passing a callback, since we're dealing with rust / bash interop. I somewhat like the port polling, actually.

In the past I would always get scolded for PRing magic numbers, but imho they can have their place if they work reasonably well and allow us to move forward with more meaningful work.

Reply:
Not that I'm recommending it, but here's [~10 lines of hacky Bash to check if a server is up](https://github.com/google/trillian/blame/master/integration/functions.sh#L34-L51)